--- a/drivers/net/wireless/ralink/rt2x00/rt2800lib.c
+++ b/drivers/net/wireless/ralink/rt2x00/rt2800lib.c
@@ -8245,7 +8245,7 @@
 	rt2800_rfcsr_write_chanreg(rt2x00dev, 60, 0xC7);
 
 	/* reduce power consumption */
-/*	rt2800_rfcsr_write_chanreg(rt2x00dev, 43, 0x53);
+	rt2800_rfcsr_write_chanreg(rt2x00dev, 43, 0x53);
 	rt2800_rfcsr_write_chanreg(rt2x00dev, 44, 0x53);
 	rt2800_rfcsr_write_chanreg(rt2x00dev, 45, 0x53);
 	rt2800_rfcsr_write_chanreg(rt2x00dev, 47, 0x64);
@@ -8258,7 +8258,7 @@
 	rt2800_rfcsr_write_chanreg(rt2x00dev, 59, 0x64);
 	rt2800_rfcsr_write_chanreg(rt2x00dev, 60, 0x4F);
 	rt2800_rfcsr_write_chanreg(rt2x00dev, 61, 0x02);
-*/
+
 	/* Initialize RF DC calibration register to default value */
 	rt2800_rfcsr_write_dccal(rt2x00dev, 0, 0x47);
 	rt2800_rfcsr_write_dccal(rt2x00dev, 1, 0x00);
